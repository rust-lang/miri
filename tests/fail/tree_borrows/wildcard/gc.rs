//@compile-flags: -Zmiri-tree-borrows -Zmiri-permissive-provenance

#[path = "../../../utils/mod.rs"]
mod utils;

#[allow(unused_variables, unused_assignments)]
fn main() {
    let mut x: u32 = 4;
    let int = {
        let y = &x;
        y as *const u32 as usize
    };
    // If y wasn't exposed, this would gc it.
    utils::run_provenance_gc();
    // This should disable y.
    x = 5;
    let wild = int as *const u32;

    let fail = unsafe { *wild }; //~ ERROR: /read access through wildcard at .* is forbidden/
}
