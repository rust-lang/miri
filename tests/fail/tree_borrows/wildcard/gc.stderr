error: Undefined Behavior: read access through wildcard at ALLOC[0x0] is forbidden
  --> tests/fail/tree_borrows/wildcard/gc.rs:LL:CC
   |
LL |     let fail=unsafe{ *wild };
   |                      ^^^^^ Undefined Behavior occurred here
   |
   = help: this indicates a potential bug in the program: it performed an invalid operation, but the Tree Borrows rules it violated are still experimental
   = help: see https://github.com/rust-lang/unsafe-code-guidelines/blob/master/wip/tree-borrows.md for further information
   = help: there are no exposed references who have read access permissions to this location
   = note: BACKTRACE:
   = note: inside `main` at tests/fail/tree_borrows/wildcard/gc.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error

